x/cork/keeper/keeper.go

```diff
@@ -222,6 +303,56 @@ func (k Keeper) GetApprovedCorks(ctx sdk.Context, threshold sdk.Dec) (approvedCo
 	return winningCorks
 }
 
+func (k Keeper) GetApprovedScheduledCorks(ctx sdk.Context, currentBlockHeight uint64, threshold sdk.Dec) (approvedCorks []types.Cork) {
+
+	corksForBlockHeight := make(map[uint64][]types.Cork)
+	var corkPowersForBlockHeight map[uint64][]int64
+
+	totalPower := k.stakingKeeper.GetLastTotalPower(ctx)
+
+	k.IterateScheduledCorks(ctx, func(val sdk.ValAddress, scheduledBlockHeight uint64, addr common.Address, cork types.Cork) (stop bool) {
+		// only operate on scheduled corksForBlockHeight that are valid
+		if currentBlockHeight < scheduledBlockHeight {
+			return false
+		}
+
+		validator := k.stakingKeeper.Validator(ctx, val)
+		validatorPower := validator.GetConsensusPower(k.stakingKeeper.PowerReduction(ctx))
+
+		found := false
+		for i, c := range corksForBlockHeight[scheduledBlockHeight] {
+			if c.Equals(cork) {
```

Comment:
The `Cork` model object itself should be the same for all signers if they are agreeing. This seems like it wouldn't necessarily map validator power to the corresponding cork, it would collect all the powers in the first matching cork.